execution/execution_test.go

```diff
@@ -146,7 +165,39 @@ func TestComplete(t *testing.T) {
 		require.NotNil(t, e)
 		require.NoError(t, err, test.name)
 		require.NotNil(t, e, test.name)
-		require.Equal(t, e.Status, Completed, test.name)
+		require.Equal(t, Completed, e.Status, test.name)
+		require.NotZero(t, e.ExecutionDuration, test.name)
+		require.Zero(t, e.Error, test.name)
+	}
+}
+
+func TestFailed(t *testing.T) {
+	e, _ := New(srv, eventID, taskKeyWithError, map[string]interface{}{"foo": "1", "bar": "2"}, tags)
+	e.Execute()
+	tests := []struct {
+		name string
+		key  string
+		data map[string]interface{}
+		xerr error
+		err  error
+	}{
+		{name: "1", key: "outputX", data: map[string]interface{}{"foo": "bar"}, xerr: errors.New("cannot complete"), err: nil},
```

Comment:
Change name "1" and "2" to the description of a testing case if you want to guide someone in tests.

Reply:
done